src/main/java/org/apache/ibatis/executor/resultset/DefaultResultSetHandler.java

```diff
@@ -540,7 +540,7 @@ private List<UnMappedColumnAutoMapping> createAutomaticMappings(ResultSetWrapper
       final List<String> unmappedColumnNames = rsw.getUnmappedColumnNames(resultMap, columnPrefix);
       // Remove the entry to release the memory
       List<String> mappedInConstructorAutoMapping = constructorAutoMappingColumns.remove(mapKey);
-      if (mappedInConstructorAutoMapping != null) {
+      if (mappedInConstructorAutoMapping != null && !mappedInConstructorAutoMapping.isEmpty()) {
```

Comment:
I am not sure why this is necessary or if it is valid.
Please add a test case that justifies the change.